lib/compress/zstd_opt.c

```diff
@@ -646,11 +657,12 @@ size_t ZSTD_compressBlock_opt_generic(ZSTD_CCtx* ctx,
                     }
                 }
 
-                ZSTD_updatePrice(optStatePtr, llen, anchor, offset, mlen);
+                ZSTD_updateStats(optStatePtr, llen, anchor, offset, mlen);
                 ZSTD_storeSeq(seqStorePtr, llen, anchor, offset, mlen-MINMATCH);
                 anchor = ip;
         }   }
-    }   /* for (cur=0; cur < last_pos; ) */
+        ZSTD_setLog2Prices(optStatePtr);
+    }   while (ip < ilimit)
```

Comment:
Is this supposed to be a comment? Currently it seems to apply to the next block after the comment, which may be confusing?

Reply:
great catch @gyscos !
Yes, it's supposed to be a comment ! fixed